main.go

```diff
@@ -299,8 +307,11 @@ func setupForwarders(ctx context.Context, hostname string, tk timekeeper.TimeKee
 	return allForwarders, nil
 }
 
+var errDupeListener = errors.New("cannot duplicate listener names or types without names")
+
 func setupListeners(tk timekeeper.TimeKeeper, hostname string, loader *config.Loader, listenFrom []*config.ListenFrom, multiplexer signalfx.Sink, logger log.Logger, scheduler *sfxclient.Scheduler) ([]protocol.Listener, error) {
 	listeners := make([]protocol.Listener, 0, len(listenFrom))
+	nameMap := make(map[string]bool)
```

Comment:
Can you not make listeners into a map indexed by name, and return the values of this map from this function?

Reply:
Could, but since the api further wants a list and there's no Values() to a go map, that would be inconvenient.